### PR TITLE
CAT-23287 Various enhancements

### PR DESCRIPTION
**Pull request by ehirsch-3pg**
1. Separated private and public subnets in lb and ecs service.
2. Added enable_cross_zone_load_balancing.
3. created the hosted_zone variable for both primary and private ones and use the is_private (default: false) variable to define how it will be used, and removed the primary_hosted_zone and private_hosted_zone
4. created variables for private and public subnets

### CODE REVIEW

**Review by raresthreepio — APPROVED** [review]

**Review by CosminG — APPROVED** [review]

File: optional.tf [hunk 1]
```diff
@@ -277,16 +289,16 @@ variable "vpc_id" {
...
+  description = "Name of the hosted zone for DNS. e.g. hosted_zone = example.org --> service.example.org. Based on the is_private, this is the primary or the private hosted zone."
   default     = null
   type        = string
 }
 
-variable "private_hosted_zone" {
-  description = "Name of the private hosted zone for DNS. e.g. private_hosted_zone = example.org --> service.example.private. If null, it is assumed that a public hosted zone will be used."
-  default     = null
-  type        = string
+variable "is_private" {
```
Comment (yhakbar) [inline optional.tf:298]:
```suggestion
variable "is_hosted_zone_private" {
```

This makes more sense to me, as it confused me why we were setting both `is_private` and `public_service` here:
https://github.com/pbs/terraform-aws-ecs-service-module/pull/93/files#diff-db0fce55d86d79a0ec71af0d6d55a52b35c4eb98271d6431fbc6054d756d8788

**Review by yhakbar — CHANGES_REQUESTED** [review]

File: locals.tf [hunk 2]
```diff
@@ -13,21 +13,22 @@ locals {
...
   hosted_zone_id                       = local.lookup_hosted_zone ? one(data.aws_route53_zone.hosted_zone[*].zone_id) : null
-  internal                             = var.internal != null ? var.internal : !var.public_service
+  internal                             = var.internal != null ? var.internal : var.is_private
   cnames                               = local.create_virtual_node ? [] : var.cnames != null ? var.cnames : [local.name]
   aliases                              = local.create_virtual_node ? [] : var.aliases != null ? var.aliases : ["${local.name}.${local.null_safe_hosted_zone}"]
   app_dns_record_count                 = local.create_lb ? length(local.cnames) : 0
   domain_name                          = !local.create_lb ? null : local.app_dns_record_count == 0 ? one(aws_lb.lb[*].dns_name) : one(aws_route53_record.app[*].fqdn)
   create_lb_sg                         = local.create_lb && var.load_balancer_type == "application"
   create_http_listeners                = local.create_lb && var.load_balancer_type == "application"
-  create_https_listeners               = local.create_lb && var.load_balancer_type == "application" && var.public_service
```
Comment (yhakbar) [inline locals.tf:30]: I think this might be problematic if we don't expect usage of Private CA to handle certs for private services.

Comment (yhakbar) [reply locals.tf:30]: While testing the module locally, I noticed the following error, which is likely related to this:

```bash
$ ./scripts/test.sh
...
                                │ Error: creating ELBv2 Listener (arn): ValidationError: A certificate must be specified for HTTPS listeners
                                │       status code: 400, request id: id
                                │ 
                                │   with module.service.aws_lb_listener.https[0],
                                │   on ../../lb.tf line 57, in resource "aws_lb_listener" "https":
                                │   57: resource "aws_lb_listener" "https" {
                                │ 
                                ╵
```

Comment (ehirsch-3pg) [reply locals.tf:30]: I will need to take a longer look on this

**Review by CosminG — APPROVED** [review]

**Review by raresthreepio — APPROVED** [review]

**Pull request merged**